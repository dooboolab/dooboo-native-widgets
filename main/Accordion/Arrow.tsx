import Svg, { Line } from 'react-native-svg';
import React from 'react';
import { View } from 'react-native';

interface Props {
  arrowPosition: string;
}

export default function Arrow(props: Props): React.ReactElement {
  return (
    <View
      style={
        props.arrowPosition === 'up' && { transform: [{ rotate: '180deg' }] }
      }>
      <Svg height="20" width="20" viewBox="0 0 20 20">
        <Line
          x1={5}
          y1={7}
          x2={10}
          y2={12}
          stroke="white"
          strokeWidth="1.5"
          strokeLinecap="round"
        />
        <Line
          x1={10}
          y1={12}
          x2={15}
          y2={7}
          stroke="white"
          strokeWidth="1.5"
          strokeLinecap="round"
        />
      </Svg>
    </View>
  );
}
