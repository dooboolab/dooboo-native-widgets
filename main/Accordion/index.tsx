import React, { FC } from 'react';

import AccrordionItem from './AccordionItem';
import { ViewStyle } from 'react-native';
import styled from 'styled-components/native';

const Container = styled.View`
  flex-direction: column;
  align-items: center;
`;

type ItemType = {
  itemTitle: string;
  itemBodies: Array<string>;
}

interface Props {
  data: Array<ItemType>;
  isAnimated?: boolean;
  collapseOnStart: boolean;
  animDuration?: number;
  activeOpacity?: number;
  titleStyle?: ViewStyle;
  itemStyle?: ViewStyle;
  itemTitleLeft?: React.ReactElement;
  itemBodyLeft?: React.ReactElement;
}

const Accordion: FC<Props> = (props) => {
  const {
    data,
    isAnimated: shouldAnimate,
    collapseOnStart,
    animDuration,
    activeOpacity,
    titleStyle,
    itemStyle,
    itemTitleLeft,
    itemBodyLeft,
  } = props;

  return (
    <Container>
      {
        data.map((itemData, idx) => {
          return (
            <AccrordionItem
              testID={`${idx}`}
              key={idx}
              itemData={itemData}
              shouldAnimate={shouldAnimate}
              collapseOnStart={collapseOnStart}
              animDuration={animDuration}
              activeOpacity={activeOpacity}
              titleStyle={titleStyle}
              itemStyle={itemStyle}
              itemTitleLeft={itemTitleLeft}
              itemBodyLeft={itemBodyLeft}
            />
          );
        })
      }
    </Container>
  );
};
export default Accordion;
