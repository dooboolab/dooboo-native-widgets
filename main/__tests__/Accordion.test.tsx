import {
  RenderResult,
  act,
  fireEvent,
  render,
} from '@testing-library/react-native';

import Accordion from '../Accordion';
import React from 'react';

// eslint-disable-next-line
let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

const createTestProps = (obj?: Record<string, unknown>): Record<string, unknown> => ({
  ...obj,
});

describe('[Accordion] renders', () => {
  it('should render without crashing', () => {
    props = createTestProps();
    component = <Accordion {...props} />;
    testingLib = render(component);
    expect(testingLib.baseElement).toBeTruthy();
  });

  it('should render when custom props given', () => {
    jest.useFakeTimers();

    props = createTestProps({
      isAnimated: true,
      visible: true,
    });
    component = <Accordion {...props} />;

    testingLib = render(component);
    expect(testingLib.baseElement).toMatchSnapshot();
  });
});

describe('[Accordion] events', () => {
  beforeEach(() => {
    props = createTestProps();
    component = <Accordion {...props} />;
    testingLib = render(component);
  });

  it('should trigger onLayout when header height changes', () => {
    const { getByTestId } = testingLib;
    const header = getByTestId('header');

    act(() => {
      fireEvent.layout(header, {
        nativeEvent: {
          layout: {
            height: 300,
          },
        },
      });
    });

    act(() => {
      fireEvent.layout(header, {
        nativeEvent: {
          layout: {
            height: 0,
          },
        },
      });
    });
  });

  it('should trigger onLayout when content height changes', () => {
    const { getByTestId } = testingLib;
    const content = getByTestId('content');

    act(() => {
      fireEvent.layout(content, {
        nativeEvent: {
          layout: {
            height: 300,
          },
        },
      });
    });

    act(() => {
      fireEvent.layout(content, {
        nativeEvent: {
          layout: {
            height: 300,
          },
        },
      });
    });
  });

  it('should set [animValue] when header and content is mounted', () => {
    const { getByTestId } = testingLib;
    const header = getByTestId('header');
    const content = getByTestId('content');

    act(() => {
      fireEvent.layout(header, {
        nativeEvent: {
          layout: {
            height: 300,
          },
        },
      });
    });

    act(() => {
      fireEvent.layout(header, {
        nativeEvent: {
          layout: {
            height: 300,
          },
        },
      });
    });

    act(() => {
      fireEvent.layout(content, {
        nativeEvent: {
          layout: {
            height: 300,
          },
        },
      });
    });

    act(() => {
      fireEvent.layout(content, {
        nativeEvent: {
          layout: {
            height: 300,
          },
        },
      });
    });
  });

  it('should trigger press event when clicking header', () => {
    act(() => {
      fireEvent.press(testingLib.getByTestId('header'));
    });

    expect(props.onPress).toBeFalsy();

    props = createTestProps({
      onPress: jest.fn(),
    });
    component = <Accordion {...props} />;
    testingLib = render(component);

    const { getByTestId } = testingLib;
    const header = getByTestId('header');

    act(() => {
      fireEvent.press(header);
    });

    expect(props.onPress).toHaveBeenCalled();
  });

  it('should trigger after custom props given', () => {
    jest.useFakeTimers();

    props = createTestProps({
      isAnimated: true,
      contentVisible: true,
    });
    component = <Accordion {...props} />;

    testingLib = render(component);

    jest.runAllTimers();

    const { getByTestId } = testingLib;
    const header = getByTestId('header');
    const content = getByTestId('content');

    act(() => {
      fireEvent.layout(header, {
        nativeEvent: {
          layout: {
            height: 300,
          },
        },
      });
    });

    act(() => {
      fireEvent.layout(content, {
        nativeEvent: {
          layout: {
            height: 300,
          },
        },
      });
    });
  });
});
