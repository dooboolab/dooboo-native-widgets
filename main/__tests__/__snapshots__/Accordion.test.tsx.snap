// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Accordion] render test should operate animation when isAnimated is true 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "paddingLeft": 40,
                "paddingRight": 20,
                "zIndex": 10,
              },
            ]
          }
          testID="itemTitle_0"
        >
          <View />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontWeight": "bold",
              }
            }
          >
            test data1
          </Text>
          <View
            style={false}
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="20"
              bbWidth="20"
              focusable={false}
              height="20"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={20}
              vbWidth={20}
              width="20"
            >
              <RNSVGGroup>
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={5}
                  x2={10}
                  y1={7}
                  y2={12}
                />
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={10}
                  x2={15}
                  y1={12}
                  y2={7}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </>
        <View
          style={
            Object {
              "height": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
            testID="itemBody_0"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "paddingLeft": 40,
                "paddingRight": 20,
                "zIndex": 10,
              },
            ]
          }
          testID="itemTitle_1"
        >
          <View />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontWeight": "bold",
              }
            }
          >
            test data2
          </Text>
          <View
            style={false}
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="20"
              bbWidth="20"
              focusable={false}
              height="20"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={20}
              vbWidth={20}
              width="20"
            >
              <RNSVGGroup>
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={5}
                  x2={10}
                  y1={7}
                  y2={12}
                />
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={10}
                  x2={15}
                  y1={12}
                  y2={7}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </>
        <View
          style={
            Object {
              "height": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
            testID="itemBody_1"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Accordion] render test should render collapsed when collapsedWhenRendered props is false 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "paddingLeft": 40,
                "paddingRight": 20,
                "zIndex": 10,
              },
            ]
          }
          testID="itemTitle_0"
        >
          <View />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontWeight": "bold",
              }
            }
          >
            test data1
          </Text>
          <View
            style={false}
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="20"
              bbWidth="20"
              focusable={false}
              height="20"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={20}
              vbWidth={20}
              width="20"
            >
              <RNSVGGroup>
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={5}
                  x2={10}
                  y1={7}
                  y2={12}
                />
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={10}
                  x2={15}
                  y1={12}
                  y2={7}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </>
        <View
          style={
            Object {
              "height": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
            testID="itemBody_0"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "paddingLeft": 40,
                "paddingRight": 20,
                "zIndex": 10,
              },
            ]
          }
          testID="itemTitle_1"
        >
          <View />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontWeight": "bold",
              }
            }
          >
            test data2
          </Text>
          <View
            style={false}
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="20"
              bbWidth="20"
              focusable={false}
              height="20"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={20}
              vbWidth={20}
              width="20"
            >
              <RNSVGGroup>
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={5}
                  x2={10}
                  y1={7}
                  y2={12}
                />
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={10}
                  x2={15}
                  y1={12}
                  y2={7}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </>
        <View
          style={
            Object {
              "height": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
            testID="itemBody_1"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Accordion] render test should render without crasing 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "paddingLeft": 40,
                "paddingRight": 20,
                "zIndex": 10,
              },
            ]
          }
          testID="itemTitle_0"
        >
          <View />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontWeight": "bold",
              }
            }
          >
            test data1
          </Text>
          <View
            style={false}
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="20"
              bbWidth="20"
              focusable={false}
              height="20"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={20}
              vbWidth={20}
              width="20"
            >
              <RNSVGGroup>
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={5}
                  x2={10}
                  y1={7}
                  y2={12}
                />
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={10}
                  x2={15}
                  y1={12}
                  y2={7}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </>
        <View
          style={
            Object {
              "height": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
            testID="itemBody_0"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "width": 300,
            },
          ]
        }
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "paddingLeft": 40,
                "paddingRight": 20,
                "zIndex": 10,
              },
            ]
          }
          testID="itemTitle_1"
        >
          <View />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontWeight": "bold",
              }
            }
          >
            test data2
          </Text>
          <View
            style={false}
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="20"
              bbWidth="20"
              focusable={false}
              height="20"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={20}
              vbWidth={20}
              width="20"
            >
              <RNSVGGroup>
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={5}
                  x2={10}
                  y1={7}
                  y2={12}
                />
                <RNSVGLine
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4294967295}
                  strokeLinecap={1}
                  strokeWidth="1.5"
                  x1={10}
                  x2={15}
                  y1={12}
                  y2={7}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </>
        <View
          style={
            Object {
              "height": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
            testID="itemBody_1"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "lightgray",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingBottom": 20,
                    "paddingLeft": 40,
                    "paddingRight": 40,
                    "paddingTop": 20,
                    "width": "100%",
                  },
                ]
              }
            >
              <View />
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontWeight": "bold",
                  }
                }
              >
                test data
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
